reset/reset.go

```diff
@@ -0,0 +1,89 @@
+package reset
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+type NodeService interface {
+	DeleteCollection(ctx context.Context, lopts metav1.ListOptions) error
+}
+
+type PersistentVolumeService interface {
+	DeleteCollection(ctx context.Context, lopts metav1.ListOptions) error
+}
+
+type PersistentVolumeClaimService interface {
+	DeleteCollection(ctx context.Context, lopts metav1.ListOptions) error
+}
+
+type StorageClassService interface {
+	DeleteCollection(ctx context.Context, lopts metav1.ListOptions) error
+}
+
+type PriorityClassService interface {
+	DeleteCollection(ctx context.Context, lopts metav1.ListOptions) error
+}
+
+type SchedulerService interface {
+	ResetScheduler() error
+}
+
+// Service cleans up resources.
+type Service struct {
+	client       clientset.Interface
+	nodeService  NodeService
+	pvService    PersistentVolumeService
+	pvcService   PersistentVolumeClaimService
+	scSerivce    StorageClassService
+	pcService    PriorityClassService
+	schedService SchedulerService
+}
```

Comment:
With [this](https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/105/files#r805167412) change, the field will be like this:

```suggestion
type Service struct {
	client       clientset.Interface
	// deleteServices has the all services for each resource.
	// key: service name. 
	deleteServices map[string]DeleteService
	schedService SchedulerService
}
```